internal/venice-common/src/main/java/com/linkedin/venice/kafka/TopicManager.java

```diff
@@ -77,64 +85,118 @@ public class TopicManager implements Closeable {
   private final String pubSubBootstrapServers;
   private final long kafkaOperationTimeoutMs;
   private final long topicMinLogCompactionLagMs;
-  private final PubSubAdminAdapterFactory<PubSubAdminAdapter> pubSubAdminAdapterFactory;
-  // TODO: Use single PubSubAdminAdapter for both read and write operations
-  private final Lazy<PubSubAdminAdapter> pubSubWriteOnlyAdminAdapter;
-  private final Lazy<PubSubAdminAdapter> pubSubReadOnlyAdminAdapter;
+  private final Lazy<PubSubAdminAdapter> pubSubDefaultAdminAdapter;
+  private final Lazy<PubSubConsumerAdapter> pubSubConsumerAdapter;
+  private final Map<String, PubSubAdminAdapter> pubSubAdminAdapterMap = new HashMap<>();
```

Comment:
Reading the type declaration of `pubSubDefaultAdminAdapter` raises a question that if it's necessary to use the `Lazy<PubSubAdminAdapter>` as the value in the map? Same for the below `pubSubConsumerAdapterMap`.

Reply:
I think if it is optional lazy is good, otherwise no need to have lazy here.